### PR TITLE
CI: add hardware test job

### PR DESCRIPTION
**Pull request by szszszsz**
Automatically flash the just built firmware to the hardware, and test for defects.

### CODE REVIEW

**Comment by szszszsz** [comment]
Perhaps it's worth removing `check-firmware` job entirely now? I guess `cargo check` was used here to avoid producing binary to save some execution time.

**Comment by robin-nitrokey** [comment]
Yes, the `check` is no longer needed but we should make sure that we `build` the same feature sets.  This means we should always build both test and stable in `build-firmware-for-tests`.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -164,3 +167,55 @@ build-usbip:
...
+  image: registry.git.nitrokey.com/nitrokey/nitrokey-3-firmware/nitrokey3:latest
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "push"'
+  tags:
+    - docker
+  stage: build
+  script:
+    - git describe
+    - export VERSION=`git describe`
+    - if echo $VERSION | grep test ; then export FEATURES=test ; fi
```
Comment (robin-nitrokey) [inline .gitlab-ci.yml:183]: Isn’t this overwritten by explicitly setting `FEATURES` in the `make` calls?

Comment (szszszsz) [reply .gitlab-ci.yml:183]: Good spot. Yes, I've added `no-button` without checking this one first. To be removed.

**Comment by robin-nitrokey** [comment]
Shouldn’t we run four checks – lpc55/nrf52 and stable/test?  Especially for the upcoming release, we are mostly interested in checking the test release.  Currently this only checks stable AFAIS.

**Pull request merged**